src/cfnlint/helpers.py

```diff
@@ -166,10 +167,63 @@
     'AWS::Redshift::Cluster'
 ]
 
+CACHING_DIR = os.path.join(os.path.dirname(__file__), 'data', 'DownloadsMetadata')
 
-def get_url_content(url):
+
+def get_url_content(url, caching=False):
```

Comment:
This function is doing too many things... it's loading up the cache, checking to see if an update needs to happen, and if it does, it is downloading the file, and then updating the cache. Can we split it up into smaller chunks?